build/cmake/functions.cmake

```diff
@@ -225,7 +225,38 @@ function(wx_set_target_properties target_name is_base)
             uxtheme
         )
         target_link_libraries(${target_name}
-            PUBLIC ${WXMSW_LIBRARIES})
+            PUBLIC ${WIN32_LIBRARIES})
+
+        if(WXGTK)
+            if(WXGTK4)
+                set(WXGTK_WIN32_LIBRARIES
+                    libgtk-4.dll.a
+                    libgdk-4.dll.a
+                )
+            elseif(WXGTK3)
+                set(WXGTK_WIN32_LIBRARIES
+                    libgtk-3.dll.a
+                    libgdk-3.dll.a
+                )
+            elseif(WXGTK2)
+                set(WXGTK_WIN32_LIBRARIES
+                    gtk-win32-2.0.lib
+                    gdk-win32-2.0.lib
+                )
+            endif()
+            set(WXGTK_WIN32_LIBRARIES ${WXGTK_WIN32_LIBRARIES}
+                gio-2.0.lib
+                pangocairo-1.0.lib
+                gdk_pixbuf-2.0.lib
+                cairo.lib
+                pango-1.0.lib
+                gobject-2.0.lib
+                gthread-2.0.lib
+                glib-2.0.lib
+            )
+            target_link_libraries(${target_name}
+                PUBLIC ${WXGTK_WIN32_LIBRARIES})
+        endif()
```

Comment:
Adding all these gtk libraries should not be necessary. They are in `wxTOOLKIT_LIBRARIES`.

Reply:
Where are these libraries in `wxTOOLKIT_LIBRARIES`? You can build without this piece of code? What cmake command line arguments you have used?
Note that wxGTK/Win32 uses Visual C++ project files instead of configure.in for build.

Reply:
I used the `MinGW Makefiles` generator instead of `Visual Studio`. I don't need the extra libraries in `WXGTK_WIN32_LIBRARIES`.

`wxTOOLKIT_LIBRARIES` is specified in `toolkit.cmake`. It contains `${gtk_lib}_LIBRARIES` (in this case `GTK2_LIBRARIES`), which is automatically set by [`find_package(GTK2)`](https://cmake.org/cmake/help/v3.1/module/FindGTK2.html).
For me it contains:
```
C:/dev/msys64/mingw64/lib/libglib-2.0.dll.a;C:/dev/msys64/mingw64/lib/libgobject-2.0.dll.a;C:/dev/msys64/mingw64/lib/libatk-1.0.dll.a;C:/dev/msys64/mingw64/lib/libgio-2.0.dll.a;C:/dev/msys64/mingw64/lib/libgthread-2.0.dll.a;C:/dev/msys64/mingw64/lib/libgmodule-2.0.dll.a;C:/dev/msys64/mingw64/lib/libgdk_pixbuf-2.0.dll.a;C:/dev/msys64/mingw64/lib/libcairo.dll.a;C:/dev/msys64/mingw64/lib/libpango-1.0.dll.a;C:/dev/msys64/mingw64/lib/libpangocairo-1.0.dll.a;C:/dev/msys64/mingw64/lib/libpangoft2-1.0.dll.a;C:/dev/msys64/mingw64/lib/libgdk-win32-2.0.dll.a;C:/dev/msys64/mingw64/lib/libgtk-win32-2.0.dll.a
```

What does it contain for you? Add `message("${wxTOOLKIT_LIBRARIES}")` after it is assigned to see its contents.

 When I use a `Visual Studio` generator, CMake cannot find all include directories and libraries:
```
Some or all of the gtk libraries were not found. (missing: GTK2_GTK_LIBRARY GTK2_GDKCONFIG_INCLUDE_DIR GTK2_GDK_LIBRARY GTK2_GLIBCONFIG_INCLUDE_DIR GTK2_GLIB_LIBRARY)
```
All the `wxTOOLKIT_` variables are left empty. Strangely it does not abort with an error, even though `REQUIRED` is specified.
 Does it correctly detect GTK2? It should print a line like:
 `Found GTK2_GTK: C:/dev/msys64/mingw64/lib/libgtk-win32-2.0.dll.a`

If you still require these changes, maybe you should wrap them with `if(MSVC)`, move it to `toolkit.cmake` and append these libraries to `wxTOOLKIT_LIBRARIES`.

Reply:
Fixed as you said, thanks.

With CMake MSVC build I have same message:
`-- Some or all of the gtk libraries were not found. (missing: GTK2_GTK_LIBRARY G
TK2_GTK_INCLUDE_DIR GTK2_GDK_INCLUDE_DIR GTK2_GDKCONFIG_INCLUDE_DIR GTK2_GDK_LIB
RARY GTK2_GLIB_INCLUDE_DIR GTK2_GLIBCONFIG_INCLUDE_DIR GTK2_GLIB_LIBRARY)`
and empty `wxTOOLKIT_LIBRARIES`. So I have added libraries manually as it is in Bakefiles.

My MinGW shipped without GTK+/Win32 libraries.